src/extensions/insights/insight.ts

```diff
@@ -0,0 +1,26 @@
+export type InsightMetaData = {
+  name: string;
+  description: string;
+};
+
+export type InsightResult = {
+  metaData: InsightMetaData;
+  message?: string;
+  data: any;
+  renderedData: any; // TOD) - should be react component
+};
+
+export type RawResult = {
+  message?: string;
+  data: any;
+};
```

Comment:
I don't think you need it. it's something internal for each insight and not must.

Reply:
I find it very convenient to have it there. Insight doesn't force the user to use it. It's there as a reference.